### PR TITLE
Incorporate function id in function call and remove `_load` method overloading

### PR DESCRIPTION
**Pull request by matzhaugen**
# Incorporate function id in function calls
When the function id is present in the call, we no longer need to overload the `_load` method which enables us to use the `_list` endpoint in the APIClient in the SDK.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,9 @@ Changes are grouped as follows
 - `Security` in case of vulnerabilities.
 
 
-## Unreleased
+## [0.7.3]
+### Changed
+- Because the new function call response from the  Cognite functions api now returns functions id, the endpoint for getting logs of a call has a simplified internal structure.
```
Comment (andeplane) [inline CHANGELOG.md:18]:
```suggestion
- Function calls now returns `functionId`, so the getting logs of a call has a simplified internal structure.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,7 +13,9 @@ Changes are grouped as follows
 - `Security` in case of vulnerabilities.
 
 
-## Unreleased
+## [0.7.3]
```
Comment (andeplane) [inline CHANGELOG.md:16]:
```suggestion
## [0.8.1]
```

File: cognite/experimental/_api/functions.py [hunk 3]
```diff
@@ -362,8 +364,8 @@ def list(self, function_id: Optional[int] = None, function_external_id: Optional
         if function_external_id:
             function_id = self._cognite_client.functions.retrieve(external_id=function_external_id).id
         url = f"/functions/{function_id}/calls"
-        res = self._get(url)
-        return FunctionCallList._load(res.json()["items"], function_id=function_id, cognite_client=self._cognite_client)
+
+        return self._list(method="POST", resource_path=url)
```
Comment (HMEiding) [inline cognite/experimental/_api/functions.py:368]: Why this change? Seems to be unrelated to this PR, right?

File: tests/tests_unit/test_api/test_functions.py [hunk 4]
```diff
@@ -87,30 +119,12 @@ def mock_functions_delete_response(rsps):
...
-    "endTime": 1585925310822,
-    "status": "Failed",
-    "error": {"message": "some message", "trace": "some stack trace"},
-}
-CALL_TIMEOUT = {"id": 5678, "startTime": 1585925306822, "endTime": 1585925310822, "status": "Timeout"}
-
-
 @pytest.fixture
 def mock_functions_call_responses(rsps):
     url = FUNCTIONS_API._get_base_url_with_base_path() + "/functions/1234/call"
```
Comment (andeplane) [inline tests/tests_unit/test_api/test_functions.py:124]: Can you update this from `1234` to `FUNCTION_ID` too, since you do it everywhere else?

Comment (matzhaugen) [reply tests/tests_unit/test_api/test_functions.py]: done

File: cognite/experimental/_api/functions.py (hunk 3 above)
Comment (matzhaugen) [reply cognite/experimental/_api/functions.py:368]: This is the very reason for this pr.

Comment (matzhaugen) [reply cognite/experimental/_api/functions.py:368]: I see, I'll remove this.

File: cognite/experimental/_api/functions.py [hunk 5]
```diff
@@ -332,6 +332,8 @@ def _validate_function_handle(function_handle):
 
 
 class FunctionCallsAPI(APIClient):
+    _LIST_CLASS = FunctionCallList
```
Comment (HMEiding) [inline cognite/experimental/_api/functions.py:335]: This is not needed, I guess?

Comment (matzhaugen) [reply cognite/experimental/_api/functions.py]: you're correct, I'll remove
